osu.Game.Rulesets.Catch/UI/Catcher.cs

```diff
@@ -464,127 +461,29 @@ private void updateState(CatcherAnimationState state)
             updateCatcher();
         }
 
-        private void placeCaughtObject(DrawablePalpableCatchHitObject drawableObject, Vector2 position)
+        private CaughtObjectEntry placeCaughtObject(DrawablePalpableCatchHitObject source, Vector2 positionInStack)
         {
-            var caughtObject = getCaughtObject(drawableObject.HitObject);
-
-            if (caughtObject == null) return;
-
-            caughtObject.CopyStateFrom(drawableObject);
-            caughtObject.Anchor = Anchor.TopCentre;
-            caughtObject.Position = position;
-            caughtObject.Scale /= 2;
+            var entry = source.HitObject is Droplet
+                ? CaughtObjectContainer.AddDropObject(source, positionInStack, DroppedObjectAnimation.Explode, stackMirrorDirection)
+                : CaughtObjectContainer.AddStackObject(source, positionInStack);
```

Comment:
This logic should probably be in `CaughtObjectContainer` (it's even duplicated in tests). Seems like there can possibly be only one method that accepts a caught HO, maybe.